src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+﻿using System;
+using System.Reactive.Disposables;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(
+        ":disabled",
+        ":secondaryButtonRight",
+        ":secondaryButtonSpan",
+        ":checkedFlyoutOpen",
+        ":flyoutOpen",
+        ":checkedTouchPressed",
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
+        ":primaryPressed",
+        ":primaryPointerOver",
+        ":secondaryPressed",
+        ":secondaryPointerOver")]
```

Comment:
It really doesn't feel right.
Usually, we either don't need custom pseudoclasses on controls or we have some small amount of helpful ones. Like "flyout-open" makes sense and we have similar pseudoclasses for combobox/autosuggestbox already.
But all of these duplicates for "checked/pressed" should be possible to replace with our selector's syntax. It makes control styling way more complex.

Reply:
Even though I understand that these pseudoclasses are basically ported from VisualStateTriggers. It should be possible to rewrite it in more Avalonia-friendly way.

Reply:
I will need to spend some time on it next time.

Reply:
I found when I originally ported this trying to translate what MS did setting all their VisualStates (see SplitButton.cpp) was really difficult to break apart and wrap into pseudoclasses because it manages two buttons, touch/mouse input, and keyboard. @robloo managed to nicely clean up what I did - but I'm sure with time, it can probably be cleaned up more

Reply:
I'm not sure that's necessary for the first version of the control. Don't let perfect be the enemy of good.

Reply:
We shouldn't care as much about future styles porting from WinUI, keeping our controls in worse state of maintability and usability.

Reply:
Well, I wouldn't say this creates a maintainability issue. Actually, it simplifies things considerably treating every state separately. If you start integrating them and chaining together I think it's going to get more complex and harder to maintain.

I agree Avalonia convention needs to be followed though and if simplification can be done here, why not. There are a lot of pseudoclasses which doesn't look quite right. However, I wouldn't hold this PR for it and it can be addressed later if you want to spend time on it.

Reply:
Was changed in later commits so resolved. Discussion continues here: https://github.com/AvaloniaUI/Avalonia/pull/7422/files#r792961686